### DISCUSSION TITLE
Backend error when executing docker compose in terminal

### DISCUSSION THREAD

**Discussion by anujnew1** [monkeytypegame/monkeytype#5245] [Questions] [answered]
![Screenshot 2024-03-23 082345](https://github.com/monkeytypegame/monkeytype/assets/74196678/8885dbf6-6f92-4216-9869-8bac35023bc1)
I am trying to self host Monkeytype I followed the [SELF_HOSTING](https://github.com/monkeytypegame/monkeytype/blob/master/SELF_HOSTING.md) when I executed `docker compose up -d` it ran smoothly and downloaded/pulled all files  but when started to run got an error `Container monkeytype-backend    Error` then I tried to execute `docker compose up -d` again got an error (you can see the image). what should I do? (I am beginner in all of this)

**Comment by fehmer** [comment]
Please provide the logs from the backend container `docker logs monkeytype-backend`.

**Reply by anujnew1** [reply]
`23-Mar-2024 02:29:49.693        No anticheat module found. Continuing in dev mode, results will not be validated.
23-Mar-2024 02:29:50.142        Starting server version DEVELOPMENT-VERSION
23-Mar-2024 02:29:50.144        Starting server in dev mode
23-Mar-2024 02:29:50.145        Connecting to database monkeytype...
23-Mar-2024 02:29:50.168        Connected to database
23-Mar-2024 02:29:50.168        Initializing Firebase app instance...
23-Mar-2024 02:29:50.172        Failed to boot server
23-Mar-2024 02:29:50.173        Service account object must contain a string "project_id" property.
FirebaseAppError: Service account object must contain a string "project_id" property.
    at new ServiceAccount (/app/node_modules/firebase-admin/lib/app/credential-internal.js:145:19)
    at new ServiceAccountCredential (/app/node_modules/firebase-admin/lib/app/credential-internal.js:70:15)
    at Object.cert (/app/node_modules/firebase-admin/lib/app/credential-factory.js:103:54)
    at init (/app/init/firebase-admin.js:29:61)
    at /app/server.js:62:39
    at Generator.next (<anonymous>)
    at fulfilled (/app/server.js:28:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errorInfo: {
    code: 'app/invalid-credential',
    message: 'Service account object must contain a string "project_id" property.'
  },
  codePrefix: 'app'
}
waiting for backend...`

when i entered the `docker logs monkeytype-backend` I got many logs like these but the time was different in every of them.

in image
![Screenshot 2024-03-24 103019](https://github.com/monkeytypegame/monkeytype/assets/74196678/9be76315-de38-4886-9fa4-4147bffea9ad)

**Comment by fehmer** [comment]
If you plan to allow login please follow the steps in [Account System](https://github.com/monkeytypegame/monkeytype/blob/master/SELF_HOSTING.md#account-system).

I'll try to figure out tomorow how to run the backend without the account system.

**Reply by anujnew1** [reply]
I followed the Account System Step and it worked. I still don't know why I encountered those errors.

**Reply by fehmer** [reply, marked as answer]
will be fixed with https://github.com/monkeytypegame/monkeytype/pull/5255
